api-sandbox/Web3.elm

```diff
@@ -1,16 +1,26 @@
-module Web3 exposing (..)
+port module Web3
+    exposing
+        ( Web3State
+        , Request
+        , Response
+        , init
+        , handleResponse
+        , getBlockNumber
+        , request
+        , response
+        )
 
 import Dict exposing (..)
 import Json.Decode exposing (..)
 
 
-type alias Model =
-    { requestMap : Dict Int (Decoder Web3Response) }
+init : Web3State msg
+init =
+    Web3State Dict.empty
 
 
-init : Model
-init =
-    { requestMap = Dict.empty }
+type Web3State msg
+    = Web3State (Dict Int (String -> msg))
```

Comment:
To clarify, the problem with just `msg` as the value type means that

```
update (msg result) model
```

can't compile in our `Web3Response` handler